workflow-bot-app/src/test/java/com/symphony/bdk/workflow/ExecuteRequestIntegrationTest.java

```diff
@@ -1,84 +1,135 @@
+        SwadlParser.fromYaml(getClass().getResourceAsStream(swadlFile));
 
-    final ApiClient mockedApiClient = mock(ApiClient.class);
-    final Map<String, String> header = Map.of("headerKey", "headerValue");
+    final Map<String, String> header = Map.of("keyOne", "valueOne", "keyTwo", "valueTwo, valueThree");
     final Map<String, Object> body = Map.of("args", Map.of("key", "value"));
-    final Map<String, Object> jsonResponse = new LinkedHashMap<>();
-    final Map<String, String> contactInnerMap = new LinkedHashMap<>();
+    final String url = "https://url.com?isMocked=true";
+    final String expectedResponse =
+        "{\"name\": \"john\",\n \"age\": \"22\",\n"
+            + "\"contact\": {\n\"phone\": \"0123456\",\n\"email\": \"john@symphony.com\"}";
+
+    final Response mockedResponse = new Response(200, expectedResponse);
+
+    when(httpClient.execute(method, url, body, header)).thenReturn(mockedResponse);
+
+    engine.deploy(workflow);
 
-    jsonResponse.put("name", "john");
-    jsonResponse.put("age", "22");
-    contactInnerMap.put("phone", "0123456");
-    contactInnerMap.put("email", "john@symphony.com");
-    jsonResponse.put("contact", contactInnerMap);
+    engine.onEvent(messageReceived("/execute"));
 
-    final ApiResponse<Object> mockedResponse = new ApiResponse<>(200, Collections.emptyMap(), jsonResponse);
+    assertThat(workflow).isExecuted()
+        .hasOutput(String.format(OUTPUTS_STATUS_KEY, workflowId), 200)
+        .hasOutput(String.format(OUTPUTS_BODY_KEY, workflowId), expectedResponse);
+  }
 
-    when(bdkGateway.apiClient(anyString())).thenReturn(mockedApiClient);
-    when(mockedApiClient.invokeAPI(eq(""), eq("GET"), anyList(), eq(body), eq(header), anyMap(), anyMap(),
-        eq("application/json"), eq("application/json"), eq(null), any(TypeReference.class)))
-        .thenReturn(mockedResponse);
+  @Test
+  void executeGetRequestSuccessful() throws Exception {
+    final Workflow workflow =
+        SwadlParser.fromYaml(getClass().getResourceAsStream("/request/execute-request-successful-GET.swadl.yaml"));
+
+    final Map<String, String> header = Map.of("keyOne", "valueOne", "keyTwo", "valueTwo, valueThree");
+    final String url = "https://url.com?isMocked=true";
+    final String expectedResponse =
+        "{\"name\": \"john\",\n \"age\": \"22\","
+            + "\n\"contact\": {\n\"phone\": \"0123456\",\n\"email\": \"john@symphony.com\"}";
+
+    final Response mockedResponse = new Response(200, expectedResponse);
+
+    when(httpClient.execute("GET", url, null, header)).thenReturn(mockedResponse);
 
     engine.deploy(workflow);
 
     engine.onEvent(messageReceived("/execute"));
 
     assertThat(workflow).isExecuted()
         .hasOutput(String.format(OUTPUTS_STATUS_KEY, "executeGetRequest"), 200)
-        .hasOutput(String.format(OUTPUTS_BODY_KEY, "executeGetRequest"), jsonResponse);
+        .hasOutput(String.format(OUTPUTS_BODY_KEY, "executeGetRequest"), expectedResponse);
```

Comment:
isn't it supposed to be a map?

Reply:
If the api returns "This is a regular String" as response, what do we put as key if it is a map?
Didn't we agree to put exactly what the api returns as a String, that can be a JSON (eventually a String/Integer which are valid JSON) or anything else. The utility function is used to process that as JSON in subsequent activities.

Reply:
my bad I'm mixing when json() is used or nut